packages/@aws-cdk/aws-stepfunctions/README.md

```diff
@@ -1,2 +1,322 @@
+
+```ts
+const submitLambda = new lambda.Function(this, 'SubmitLambda', { ... });
+const getStatusLambda = new lambda.Function(this, 'CheckLambda', { ... });
+
+const submitJob = new stepfunctions.Task(this, 'Submit Job', {
+    resource: submitLambda,
+    resultPath: '$.guid',
+});
+
+const waitX = new stepfunctions.Wait(this, 'Wait X Seconds', { secondsPath: '$.wait_time' });
+
+const getStatus = new stepfunctions.Task(this, 'Get Job Status', {
+    resource: getStatusLambda,
+    inputPath: '$.guid',
+    resultPath: '$.status',
+});
+
+const jobFailed = new stepfunctions.Fail(this, 'Job Failed', {
+    cause: 'AWS Batch Job Failed',
+    error: 'DescribeJob returned FAILED',
+});
+
+const finalStatus = new stepfunctions.Task(this, 'Get Final Job Status', {
+    resource: getStatusLambda,
+    inputPath: '$.guid',
+});
+
+const definition = submitJob
+    .next(waitX)
+    .next(getStatus)
+    .next(new stepfunctions.Choice(this, 'Job Complete?')
+        .on(stepfunctions.Condition.stringEquals('$.status', 'FAILED'), jobFailed)
+        .on(stepfunctions.Condition.stringEquals('$.status', 'SUCCEEDED'), finalStatus)
+        .otherwise(waitX));
+
+new stepfunctions.StateMachine(this, 'StateMachine', {
+    definition,
+    timeoutSeconds: 300
+});
+```
+
+## State Machine
+
+A `stepfunctions.StateMachine` is a resource that takes a state machine
+definition. The definition is specified by its start state, and encompasses
+all states reachable from the start state:
+
+```ts
+const startState = new stepfunctions.Pass(this, 'StartState');
+
+new stepfunctions.StateMachine(this, 'StateMachine', {
+    definition: startState
+});
+```
+
+State machines execute using an IAM Role, which will automatically have all
+permissions added that are required to make all state machine tasks execute
+properly (for example, permissions to invoke any Lambda functions you add to
+your workflow). A role will be created by default, but you can supply an
+existing one as well.
+
+## Amazon States Language
+
+This library comes with a set of classes that model the [Amazon States
+Language](https://states-language.net/spec.html). The following State classes
+are supported:
+
+* `Task`
+* `Pass`
+* `Wait`
+* `Choice`
+* `Parallel`
+* `Succeed`
+* `Fail`
+
+An arbitrary JSON object (specified at execution start) is passed from state to
+state and transformed during the execution of the workflow. For more
+information, see the States Language spec.
+
+### Task
+
+A `Task` represents some work that needs to be done. It takes a `resource`
+property that is either a Lambda `Function` or a Step Functions `Activity`.
+
+```ts
+const task = new stepfunctions.Task(this, 'Invoke The Lambda', {
+    resource: myLambda,
+    inputPath: '$.input',
+    timeoutSeconds: 300,
+});
+
+// Add a retry policy
+task.retry({
+    intervalSeconds: 5,
+    maxAttempts: 10
+});
+
+// Add an error handler
+task.onError(errorHandlerState);
+
+// Set the next state
+task.next(nextState);
```

Comment:
Yeah, maybe. That's a bigger departure from the terminology used in Step Functions itself though. I think people are bound to think in term of `next`.

The upstream terminology for `onError` is called `catch`, but that's definitely a keyword in many programming languages, so I didn't want to use that.

Reply:
Seems like a state may have multiple [Fallback States](https://docs.aws.amazon.com/step-functions/latest/dg/amazon-states-language-errors.html#amazon-states-language-fallback-states), and one can also specify `ErrorEquals` and `ResultsPath` fields. How are these specified for `onError`?

Perhaps, `addCatch` and `setNext` would work instead of `next` and `onError`?

Reply:
Maybe `nextState` and `addCatchState`...

I am fine with `next` and `onError` if you feel strongly.

Reply:
Punted to SFN.

Reply:
How about `nextState(state)` and maybe `catchState(state)`. Not ideal, but at least consistent